src/scripts/pbs_status.py

```diff
@@ -377,23 +377,21 @@ def get_finished_job_stats(jobid):
     
 
 _qstat_location_cache = None
-def get_qstat_location():
+def get_pbs_location(program):
     """
-    Locate the copy of qstat the blahp configuration wants to use.
+    Locate the copy of the PBS bin the that the blahp configuration wants to use.
     """
     global _qstat_location_cache
     if _qstat_location_cache != None:
         return _qstat_location_cache
 
-    cmd = 'echo "%s/%s"' % (config.get('pbs_binpath'), 'qstat')
+    pbs_bindir = config.get('pbs_binpath')
```

Comment:
Presumably the "most offensive" bit was shelling out to run `echo`.

I don't know why that was done, but presumably, since the actual path string is constructed as an argument to `echo`, the only reason to run it through the shell would be for something like shell variable expansion.

Eg, if `pbs_binpath` was something like `$HOME/pbs` then the frightening `echo` thing would at least provide variable substitution.  Is it known whether or not any users might be relying on this feature?

Reply:
I don't think we advertise that we support setting BLAH config with env vars but it's a good point so I'm going to grudgingly close this. Could you partially revert https://github.com/htcondor/BLAH/pull/5 to restore the `echo` grossness?

Reply:
Hrm... Oh i forgot about that we did that for the other script also.  You know, it looks like the echo thing was added to both scripts in 9bfeb4f46e0 ([here](https://github.com/htcondor/BLAH/commit/9bfeb4f46e0#diff-eb4e619f061f2e2bb73dcb8bfadf09a6R363) and [here](https://github.com/htcondor/BLAH/commit/9bfeb4f46e0#diff-1ed26f5a37a1453b4844b84573326296R341)), which, as i guess you were hinting at in #5, was actually your commit.

So in a way that answers the question about "why it was done" -- that is, if it was your commit and it wasn't your intention to support variable substitution (though kind of a neat accidental feature! in classic blah style).  And, as you said, we don't advertise the feature, then someone would have had to accidentally stumble upon the fact that variable substitution worked.

... So given that it was never intended to be a feature and it was never advertised as one, i don't actually object to removing the echo thing (unless you really like the feature), provided that we make at least a small note about it in the next blahp release.

Reply:
> So in a way that answers the question about "why it was done" -- that is, if it was your commit and it wasn't your intention to support variable substitution (though kind of a neat accidental feature! in classic blah style). And, as you said, we don't advertise the feature, then someone would have had to accidentally stumble upon the fact that variable substitution worked.

Since you brought it up in this PR, I remembered deciding to shell out to `echo` specifically for the var substitution in https://github.com/htcondor/BLAH/commit/9bfeb4f46e00dee8ea8086ff7dabdb6787bdcdf8. Let's keep it.

Reply:
Ah, ok.  Rats, i guess i guessed right.

Reply:
A note for later - 

If we want to avoid the perils of the shell, another option is to do a limited environment subst right in python, eg

```
def envmatch(m):
    return os.getenv(m.group(1) or m.group(2)) or ""

# substitute occurences of $VAR or ${VAR} in path
path = re.sub(r'\$(?:{(\w+)}|(\w+))', envmatch, path)
```